src/Ejector.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.9;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {IEjector} from "./interfaces/IEjector.sol";
+import {IRegistryCoordinator} from "./interfaces/IRegistryCoordinator.sol";
+import {IStakeRegistry} from "./interfaces/IStakeRegistry.sol";
+import {BitmapUtils} from "./libraries/BitmapUtils.sol";
+
+/**
+ * @title Used for automated ejection of operators from the RegistryCoordinator
+ * @author Layr Labs, Inc.
+ */
+contract Ejector is IEjector, Ownable{
+
+    /// @notice The basis point denominator for the ejectable stake percent
+    uint16 internal constant BIPS_DENOMINATOR = 10000;
+
+    IRegistryCoordinator public immutable registryCoordinator;
+    IStakeRegistry public immutable stakeRegistry;
+
+    /// @notice Address permissioned to eject operators under a ratelimit
+    address public ejector;
+
+    /// @notice Keeps track of the total stake ejected for a quorum within a time delta
+    mapping(uint8 => StakeEjection[]) public stakeEjectedForQuorum;
+    /// @notice Ratelimit parameters for each quorum
+    mapping(uint8 => QuorumEjectionParams) public quorumEjectionParams;
+
+    constructor(
+        address _owner, 
+        address _ejector,
+        IRegistryCoordinator _registryCoordinator, 
+        IStakeRegistry _stakeRegistry,
+        QuorumEjectionParams[] memory _quorumEjectionParams
+    ) {
+        registryCoordinator = _registryCoordinator;
+        stakeRegistry = _stakeRegistry;
+        _transferOwnership(_owner);
+        _setEjector(_ejector);
+
+        for(uint8 i = 0; i < _quorumEjectionParams.length; i++) {
+            quorumEjectionParams[i] = _quorumEjectionParams[i];
+        }
+    }
+
+    /**
+     * @notice Ejects operators from the AVSs registryCoordinator
+     * @param _operatorIds The addresses of the operators to eject
+     * @param _quorumBitmaps The quorum bitmaps for each respective operator
+     */
+    function ejectOperators(bytes32[] memory _operatorIds, uint256[] memory _quorumBitmaps) external {
```

Comment:
i prefer something like
`uint256 quorumBitmap, bytes32[quorumNumbers][numOperators] operatorIds`
then we sum up the stake of operators for each quorum and boot them out if low enough and time delta has passed.

seems much simpler and just need to store the previous ejection time